src/icmp.c

```diff
@@ -61,7 +61,16 @@ static int icmp_echo(struct rte_mbuf *mbuf)
         goto errout;
 
     if (rte_raw_cksum(ich, mbuf->pkt_len) != 0xffff) {
-        RTE_LOG(WARNING, ICMP, "%s: bad checksum\n", __func__);
+        char sbuf[64], dbuf[64];
+        const char *saddr, *daddr;
+
+        saddr = inet_ntop(AF_INET, &iph->src_addr,
+                          sbuf, sizeof(sbuf)) ? sbuf : "::";
+        daddr = inet_ntop(AF_INET, &iph->dst_addr,
+                          dbuf, sizeof(dbuf)) ? dbuf : "::";
+
+        RTE_LOG(WARNING, ICMP, "%s: %s->%s, bad checksum\n",
+                __func__, saddr, daddr);
```

Comment:
If checksum failed, is it safe to extract src/dst ip addresses from mbuf.

Reply:
Here is the output.

Jul 27 20:22:49 n20-046-013 ttgw[1197794]: ICMP: icmp_echo: 112.86.64.103->220.213.141.21, bad checksum